source/_posts/2025-05-29-ai-assisted-coding.markdown

```diff
@@ -0,0 +1,188 @@
+---
+title: AI-assisted coding for teams that can't get away with vibes
+kind: article
+author: Atharva Raykar
+created_at: 2025-05-29 00:00:00 UTC
+layout: post
+---
+*Status: Living document based on production experience*
+*Last updated: 5-Jun-2025*
+
+AI should be adopted by serious engineering teams that want to build thoughtful, well-crafted products. This requires skillful usage of these tools. Our obsession with building high-quality software for over a decade has driven us to figure out how this new way of building can result in better products.
+
+This living playbook is based on our experience working with AI tools in the messy trenches of production software, where no one can afford to get away with vibes. I hope other teams can learn and benefit from our findings.
+
+## Why build with AI?
+
+**Building with AI is fast**. With the right kind of usage, I have indeed seen massive improvements in velocity for shipping new features. The velocity improvements are not evenly spread. In our experience, the smaller straightforward tasks benefit massively more from AI compared to bigger, gnarlier work such as designing a good architecture for a system. But the speed-up in the smaller ongoing engineering work helps teams allocate more time and focus on harder problems. Moreover, when velocity increases at an individual level, you can eliminate coordination overhead with leaner teams, which can unlock even more velocity.
```

Comment:
This section can be much shorter or come at the end. The tips & tricks below seem more valuable. This para is not specific, so it reads a bit like filler (for example what is "small" or "big" task – sounds a bit trite without an example).

Reply:
I gave an example of big task? (designing a good architecture for a system).
can add a similar example for small task (adding a CRUD endpoint)

I can't think of what to eliminate here. I want to say why AI tooling is worth adopting, don't want to assume the reader already knows this.

In fact this is basically the biggest blocker and motivator for the post. I always have felt a lot of labour required to tell people to try it properly. Most people aren't really that sold by default.